src/include/crc32c.h

```diff
@@ -26,7 +40,13 @@ extern ceph_crc32c_func_t ceph_choose_crc32(void);
  */
 static inline uint32_t ceph_crc32c(uint32_t crc, unsigned char const *data, unsigned length)
 {
+  if (!data && length > 16)
```

Comment:
Your algorithm makes use of large precomputed array (coincidentally, that is 4 kilobytes of data). Your code will cause cache misses across these 4KB, which *may* be as costly (or more costly) as actually computing crc across up to 4KB of actual data (the cost of cache miss in that case will be balanced by cache prefetcher). So that's why I agree on putting "length >= 2048" at least.